### ISSUE TITLE
sozuctl shutdown returns immediatly, without waiting for workers to exit

### ISSUE THREAD

**Issue by BlackYoup** [sozu-proxy/sozu#718]
Using Sozu 0.14, `sozuctl shutdown` will make the main process quit while keeping the workers alive until all connections end. `sozuctl shutdown` will also immediately return, making the user think that all connections have been closed while they weren't.

`sozuctl` can't be used after that because the socket refuses connections. Workers don't accept new connections, as they should.

To reproduce:
- Start a sozu instance with a configured backend that answers after 10 seconds
- curl the proxy, the request should hang for 10 seconds waiting for the backend to answer
- issue a `sozuctl shutdown`, it should return immediately
- your curl request should end successfully a few seconds after
- listing running sozu processes using `ps aux` shows that only the worker is still there while the main process has quit

**Comment by Keksoj** [comment]
I reproduced the issue step by step but on my machine, `sozuctl shutdown` hangs until the lagging backend has replied, which seems to be the desired outcome.

Some time has passed and many things have been merged since this issue has been posted, so maybe that's the cause.

After sending a shutdown order to the workers, the code seems to wait for an OK response from all of them before shutting down the main process with a Masterstop.

I'm slightly refactoring and **commenting** the code anyway, maybe I'll stumble upon something weird.

I'm
